### ISSUE TITLE
CORe50Dataset return the whole dataset (train + test set) with the attribute train=True

### ISSUE THREAD

**Issue by ggraffieti** [ContinualAI/avalanche#928]
🐛 **Describe the bug**
The CORe50Dataset return the whole dataset (train + test set) with the attribute train=True

🐜 **To Reproduce**
```python
core50_train = CORe50Dataset(
    root="../data/dataset/core50",
    train=True,
    mini=False,
    object_level=True
)
print(len(core50_train)   # should be ~120k, is more than 160k (whole dataset)
```

🐝 **Expected behavior**
Only the training set should be returned.

🐞 **Screenshots**
N/A

🦋 **Additional context**
N/A

**Comment by vlomonaco** [comment]
OpenLoris dataset should be checked as well!
